.circleci/config.yml

```diff
@@ -0,0 +1,169 @@
+version: 2.1
+
+orbs:
+  aws-ecs: circleci/aws-ecs@1.2.0
+  docker: circleci/docker@1.2.1
+  service-adapter:
+    commands:
+      ecr_login:
+        steps:
+          - run:
+              name: Login to ECR
+              command: |
+                temp_role=$(aws sts assume-role --role-arn arn:aws:iam::902837325998:role/ccms-ecr-role --role-session-name ci)
+                export AWS_ACCESS_KEY_ID=$(echo $temp_role | jq .Credentials.AccessKeyId | xargs)
+                export AWS_SECRET_ACCESS_KEY=$(echo $temp_role | jq .Credentials.SecretAccessKey | xargs)
+                export AWS_SESSION_TOKEN=$(echo $temp_role | jq .Credentials.SessionToken | xargs)
+                eval $(aws ecr get-login --no-include-email --region=eu-west-2)
+      aws-credentials:
+        steps:
+          - run:
+              name: Set AWS credentials
+              command: |
+                temp_role=$(aws sts assume-role --role-arn arn:aws:iam::$AWS_ACCOUNT:role/ccms-service-role --role-session-name ciservice)
+                export AWS_ACCESS_KEY_ID=$(echo $temp_role | jq .Credentials.AccessKeyId | xargs)
+                export AWS_SECRET_ACCESS_KEY=$(echo $temp_role | jq .Credentials.SecretAccessKey | xargs)
+                export AWS_SESSION_TOKEN=$(echo $temp_role | jq .Credentials.SessionToken | xargs)
+                export AWS_REGION="eu-west-2"
+                echo "export AWS_ACCESS_KEY_ID=$AWS_ACCESS_KEY_ID" >> $BASH_ENV
+                echo "export AWS_SECRET_ACCESS_KEY=$AWS_SECRET_ACCESS_KEY" >> $BASH_ENV
+                echo "export AWS_SESSION_TOKEN=$AWS_SESSION_TOKEN" >> $BASH_ENV
+                echo "export AWS_REGION=$AWS_REGION" >> $BASH_ENV
+                aws configure set aws_access_key_id $AWS_ACCESS_KEY_ID
+                aws configure set aws_secret_access_key $AWS_SECRET_ACCESS_KEY
+                aws configure set region $AWS_REGION --profile default
+
+jobs:
+  build:
+    working_directory: ~/laa-ccms-service-adapter
+    docker:
+      - image: circleci/openjdk:8-jdk-browsers
+    steps:
+      - checkout
+      - add_ssh_keys
+      - run: ./mvnw clean install
+      - persist_to_workspace:
+          root: ~/laa-ccms-service-adapter
+          paths:
+            - target/assess-service-adapter.jar
+
+  build-and-push-docker-image:
+    executor: docker/machine
+    steps:
+      - checkout
+      - attach_workspace:
+          at: /tmp/workspace
+      - run:
+          name: Install aws cli
```

Comment:
Given this lib is a pre-req to almost all docker images needed when working with AWS envs, should this be pre-baked in to the docker image requested?

Reply:
I'm not sure what you mean by the docker image requested? CircleCI has [an orb](https://circleci.com/developer/orbs/orb/circleci/aws-cli) for AWS CLI we could use so that we don't have to install it twice (not sure if that's what you mean though?)

Reply:
I am not actually familiar with how circle CI runtime environment is built, but there may be a place where you could specify to use the runtime env using an image. That image could be prepacked with AWS cli if that makes sense? Again this isn't a deal breaker.

Reply:
I think that is possible, but might be more complicated than what we have. For each job in CircleCI you specify the environment it runs in, so for this job we need to have Docker installed and we need AWS CLI to log in to ECR registry. We could create a custom image which has AWS CLI installed and can run Docker commands, but I think what we have is simpler